desc/geometry/curve.py

```diff
@@ -533,3 +630,243 @@ def set_coeffs(self, n, r=None):
+
+        Returns
+        -------
+        curve : FourierXYZCurve
+            New representation of the curve parameterized by Fourier series for X,Y,Z.
+
+        """
+        coords = self.compute("x", grid=grid, basis="xyz")["x"]
+        return FourierXYZCurve.from_values(coords, N=N, basis="xyz", name=name)
+
+    def to_SplineXYZCurve(self, knots=None, grid=None, method="cubic2", name=""):
+        """Convert Curve to SplineXYZCurve.
+
+        Parameters
+        ----------
+        knots : ndarray
+            arbitrary theta values to use for spline knots,
+            should be an 1D ndarray of same length as the input.
+            (input length in this case is determined by grid argument, since
+            the input coordinates come from
+            Curve.compute("x",grid=grid))
+            If None, defaults to using an equal-arclength angle as the knots
+            If supplied, will be rescaled to lie in [0,2pi]
+        grid : Grid, int or None
+            Grid used to evaluate curve coordinates on to fit with SplineXYZCurve.
+            If an integer, uses that many equally spaced points.
+        method : str
+            method of interpolation
+            - `'nearest'`: nearest neighbor interpolation
+            - `'linear'`: linear interpolation
+            - `'cubic'`: C1 cubic splines (aka local splines)
+            - `'cubic2'`: C2 cubic splines (aka natural splines)
+            - `'catmull-rom'`: C1 cubic centripetal "tension" splines
+        name : str
+            name for this curve
+
+        Returns
+        -------
+        SplineXYZCurve: SplineXYZCurve
+            New representation of the curve parameterized by a spline for X,Y,Z.
+
+        """
+        coords = self.compute("x", grid=grid, basis="xyz")["x"]
+        return SplineXYZCurve.from_values(
+            coords, knots=knots, method=method, name=name, basis="xyz"
+        )
+
+
+class SplineXYZCurve(Curve):
+    """Curve parameterized by spline knots in X,Y,Z.
+
+    Parameters
+    ----------
+    X, Y, Z: array-like
+        points for X, Y, Z describing a closed curve
+    knots : ndarray
+        arbitrary theta values to use for spline knots,
+        should be an 1D ndarray of same length as the input X,Y,Z.
+        If None, defaults to using an equal-arclength angle as the knot
+        If supplied, will be rescaled to lie in [0,2pi]
+    method : str
+        method of interpolation
+        - `'nearest'`: nearest neighbor interpolation
+        - `'linear'`: linear interpolation
+        - `'cubic'`: C1 cubic splines (aka local splines)
+        - `'cubic2'`: C2 cubic splines (aka natural splines)
+        - `'catmull-rom'`: C1 cubic centripetal "tension" splines
+    name : str
+        name for this curve
+
+    """
+
+    _io_attrs_ = Curve._io_attrs_ + ["_X", "_Y", "_Z", "_basis", "_transform"]
+
+    def __init__(
+        self,
+        X,
+        Y,
+        Z,
+        knots=None,
+        method="cubic",
```

Comment:
for the SplineXYZCoil and all the `to_SplineXYZCurve` methods the default method is `cubic2`, any reason its different here?

Reply:
I had ran into some issues when loading from a MAKEGRID file where cubic2 interpolation failed at the endpoints and gave NaN, but cubic did not have that same issue